platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/internal/service/scopes/Scope.java

```diff
@@ -17,75 +17,131 @@
      * <p>
-     * Generally, one regular Gradle invocation is conceptually a session.
-     * However, the GradleBuild task is currently implemented in such a way that it uses a discrete session.
-     * Having the GradleBuild task reuse the outer session is complicated because it <a href="https://github.com/gradle/gradle/issues/4559">may use a different Gradle user home</a>.
+     * A regular Gradle invocation requires only one build session.
+     * However, when the {@code GradleBuild} task is involved, it can create "nested" build sessions.
+     * Having the {@code GradleBuild} task reuse the outer session is complicated because
+     * it <a href="https://github.com/gradle/gradle/issues/4559">can use a different Gradle user home</a>.
+     * <p>
+     * The cross build session state is managed by the {@code CrossBuildSessionState} class.
+     * An instance is created per Gradle invocation.
      *
-     * <p>{@link Global} and parent scope services are visible to {@link CrossBuildSession} scope services, but not vice versa.</p>
+     * <p>{@link Global} services are visible to {@link CrossBuildSession} services and descendant scopes, but not vice versa.
      */
     interface CrossBuildSession extends Global {}
 
     /**
-     * These services are reused across build invocations in a session.
-     *
-     * A build session can be long-lived in a continuous build (where these services would be reused) or short-lived in a
-     * regular, single build.
-     *
-     * <p>{@link UserHome}, {@link CrossBuildSession} and parent scope services are visible to {@link BuildSession} scope services, but not vice versa.</p>
+     * The scope of a build session.
+     * <p>
+     * A <em>build session</em> represents a single invocation of Gradle, for example when you run {@code gradlew build}.
+     * A session runs the build one or more times.
+     * For example, when continuous build is enabled, the session may run the build many times,
+     * but when it is disabled, the session will run the build only once.
+     * <p>
+     * The build session state is managed by the {@code BuildSessionState} class.
+     * An instance is created at the start of a Gradle invocation and discarded at the end of that invocation.
+     * <p>
+     * {@link UserHome}, {@link CrossBuildSession} and their parent services are visible to
+     * {@link BuildSession} services and descendant scopes, but not vice versa.
      */
     interface BuildSession extends UserHome, CrossBuildSession {}
 
     /**
-     * These services are recreated when in continuous build and shared across all nested builds.
-     * They are closed when the build invocation is completed.
-     *
-     * <p>{@link BuildSession} and parent scope services are visible to {@link BuildTree} scope services, but not vice versa.</p>
+     * The scope of a single <em>build execution</em> within a {@link BuildSession build session}.
```

Comment:
> So, the "build tree" scope exists due to continuous builds? If we wouldn't have such a thing as a continuous build, then this scope wouldn't be needed?

Not exactly. `BuildSession` and `CrossBuildSession` exists because of the continuous builds, to share some state between iterations.

> The term of "tree of builds" also makes me think of included builds, but they don't have anything to do with this scope, do they?

They do. A composite build is a `BuildTree`, and each of its included builds is a `Build`. The root build is also a `Build` for that matters.